File: Java/Lista 3/L3_Ex1.1/src/Estante.java

import java.util.ArrayList;


public class Estante {
    private String nome; 
    private CategoriaEnum categoria; 
    private ArrayList<Livro> livros = new ArrayList();

    public Estante(String nome, CategoriaEnum categoria) {
        this.nome = nome;
        this.categoria = categoria;
    }

    public String getNome() {
        return nome;
    }

    public CategoriaEnum getCategoria() {
        return categoria;
    }
    
    public void inserirLivro(Livro livro){
        livros.add(livro);
    }
    
    public void removerLivro(Livro livro){
        livros.remove(livro);
    }
    public Livro getLivros(int i){
      return livros.get(i);
    }
    
    public int gtdLivros(){
        return livros.size();
    }
}

File: Swift/L4/Lista_04_EX02/Estante.swift
public class Estante {
    private nome: String 
    private categotia: CategoriaEnum 
    private ArrayLivros = [Livro]() 

    init(nome: String,categoria: CategoriaEnum) {
        self.nome = nome
        self.categoria = categoria
    }

    func getNome() -> String{
        return nome
    }

    func getCategoria() -> CategoriaEnum {
        return categoria
    }
    
    func inserirLivro(livro: Livro){
        ArrayLivros.append(livro)
    }
    
    func removerLivro(livro: Livro){
        ArrayLivros.remove(livro)
    }
    
    func getLivros(i: int) -> Livro{
      return ArrayLivros.get(at: i)
    }
}